jetty-ee9/jetty-ee9-webapp/src/test/java/org/acme/webapp/GetRealPathsServlet.java

```diff
@@ -0,0 +1,49 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.acme.webapp;
+
+import java.io.IOException;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+public class GetRealPathsServlet extends HttpServlet
+{
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException
+    {
+        collectResourcePaths("/").stream()
+            .map(p -> getServletContext().getRealPath(p))
+            .forEach(resp.getWriter()::println);
+        resp.getWriter().flush();
+    }
+
+    private Set<String> collectResourcePaths(String path)
+    {
+        Set<String> allResourcePaths = new LinkedHashSet<>();
+        Set<String> pathsForPath = getServletContext().getResourcePaths(path);
```

Comment:
I think you meant to call `getServletContext().getRealPath(path)` here. 

You should probably rename this method and variables for clarity too.

Reply:
This is doing the right thing.
It collects all of the servlet references first with `getResourcePaths()`, then iterates through each result and uses `getRealPath()` against it.
Outputting the `getRealPath()` of each result to the `HttpServletResponse.getWriter()`.

Besides, you wrote this, for ee10, and this is just copied into place for ee9. ;-)

Reply:
You are right, I mis-read the code.